includes/validate.php

```diff
@@ -263,14 +263,52 @@ function edac_get_content( $post ) {
 	$username = get_option( 'edacp_authorization_username' );
 	$password = get_option( 'edacp_authorization_password' );
 
+
+	// Check if server returns that the domain IP is a local/loopback address.
+	// If so then file_get_contents calls from this server to this domain will
+	// likely not be able to verify ssl. So we need to use a context that
+	// does not try to validate the ssl, otherwise file_get_contents will fail.
+	// See: https://www.php.net/manual/en/context.ssl.php .
+
+	$no_verify_ssl = false; // Verify by default.
+
+	$is_local_loopback = get_option( 'edac_local_loopback', null );
+	
+	if ( null === $is_local_loopback ) {
+
+		$parsed_url = wp_parse_url( home_url() );
+
+		if ( isset( $parsed_url['host'] ) ) {
+			
+			// Get the ip of the domain.
+			$site_domain = $parsed_url['host'];
+			$host_ip     = gethostbyname( $site_domain );
```

Comment:
Do we need to be concerned with IPv6 as `gethostbyname()` doesn't support IPv6 addresses? `dns_get_record()` might be an alternative function if it's something we should be concerned with.